picire/outcome_cache.py

```diff
@@ -131,7 +131,7 @@ def lookup(self, config):
         return self.container.get(self.test_builder(config), None)
 
     def __str__(self):
-        return '{\n%s}' % ''.join('\t%r: %r,\n' % (k, v.name) for k, v in sorted(self.container.items()))
+        return '{\n%s}' % ''.join(f'\t{k!r}: {v.name!r},\n' for k, v in sorted(self.container.items()))
```

Comment:
Here, the use of `%` operator remained. Couldn't this be written as
```py
return '{{\n{"".join(f"\t{k!r}: {v.name!r},\n" for k, v in sorted(self.container.items()))}}}'
```
?

Reply:
Yes, it works. I was thinking about such transformation but it looked a bit complicate for the first sight. For the second sight, it's fine though.

Reply:
It seems that this will be the solution at the end, since we cannot put backslashes into the expression part of f-strings.